include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -155,7 +162,8 @@ lower_bound_impl(__internal::__hetero_tag<_BackendTag>, Policy&& policy, InputIt
     const bool use_32bit_indexing = size <= std::numeric_limits<std::uint32_t>::max();
     __bknd::__parallel_for(
         _BackendTag{}, ::std::forward<decltype(policy)>(policy),
-        custom_brick<StrictWeakOrdering, decltype(size), search_algorithm::lower_bound>{comp, size, use_32bit_indexing},
+        custom_brick<StrictWeakOrdering, decltype(size), decltype(zip_vw), search_algorithm::lower_bound>{
```

Comment:
I'd suggest we do the same for custom_brick as well, adding a constructor rather than passing `{}`.

Reply:
Added constructor to custom_brick and adjusted binary search, upper bound, and lower bound usages to remove `{}`.